ext/RMagick/rmimage.c

```diff
@@ -861,10 +935,17 @@ crisscross(int bang, VALUE self, Image *fp(const Image *, ExceptionInfo *))
  * @return a new image
  */
 static VALUE
+#if defined(IMAGEMAGICK_7)
+auto_channel(int argc, VALUE *argv, VALUE self, MagickBooleanType (*fp)(Image *, ExceptionInfo *exception))
```

Comment:
Like `adaptive_channel_method`, it is better to use #if ~ #else ~ # endif only where it is needed.